### PR TITLE
cefsrc: Enable definition of cef log severity

### PR DESCRIPTION
**Pull request by aiden-loroco**
Allows users to define the log_severity setting in cef to `verbose / debug | info | warning | error | fatal | disable[default]` (corresponding to `cef_log_severity_t` options [here](https://magpcss.org/ceforum/apidocs3/projects/(default)/cef_types.h.html)).

### CODE REVIEW

File: gstcefsrc.cc [hunk 1]
```diff
@@ -2,12 +2,8 @@
 #include <iostream>
 #include <sstream>
 
-#include <include/cef_app.h>
```
Comment (MathieuDuponchelle) [inline gstcefsrc.cc:5]: I can't tell if the modified includes have any impact, but they shouldn't be part of this commit in any case :)

File: gstcefsrc.cc [hunk 2]
```diff
@@ -830,6 +844,12 @@ gst_cef_src_class_init (GstCefSrcClass * klass)
           "(Example: --noexpose_wasm --expose-gc)",
           NULL, (GParamFlags) (G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS | GST_PARAM_MUTABLE_READY)));
 
+  g_object_class_install_property (gobject_class, PROP_LOG_SEVERITY,
+    g_param_spec_string ("log-severity", "log-severity",
```
Comment (MathieuDuponchelle) [inline gstcefsrc.cc:848]: You want to register an enum here, and install the property as `g_param_spec_enum` :)

**Comment by MathieuDuponchelle** [comment]
Thanks, that seems useful, but I don't think the PR is quite right yet :)

**Comment by sidsethupathi** [comment]
From #48 
> Is this related to #49 or is the topic similarity coincidental? :)

I think it's coincidental :)

@aiden-loroco and I were trying to get `cefsrc` running in a Docker container but the `gst-launch` command kept exiting without displaying anything useful (it just logged out `INFO cefsrc gstcefsrc.cc:423:run_cef: Initializing CEF` and then exited).

Once we turned `settings.log_severity = LOGSEVERITY_DEBUG`, we got something that helped us figure out we needed to set `sandbox=false` (`Running as root without --no-sandbox is not supported. See https://crbug.com/638180.`).

File: gstcefsrc.cc (hunk 1 above)
Comment (aiden-loroco) [reply gstcefsrc.cc:5]: These are already included in the header. I'll split out into another commit.

File: gstcefsrc.cc (hunk 2 above)
Comment (aiden-loroco) [reply gstcefsrc.cc]: Nice, that's much neater. I should've known there was gobject way!

File: gstcefsrc.h [hunk 3]
```diff
@@ -1,6 +1,8 @@
 #ifndef __GST_CEF_SRC_H__
 #define __GST_CEF_SRC_H__
 
+#include <glib.h>
```
Comment (MathieuDuponchelle) [inline gstcefsrc.h:4]: That shouldn't be needed, `gst.h` will bring it in

File: gstcefsrc.cc [hunk 4]
```diff
@@ -824,6 +858,12 @@ gst_cef_src_class_init (GstCefSrcClass * klass)
           "(Example: --noexpose_wasm --expose-gc)",
           NULL, (GParamFlags) (G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS | GST_PARAM_MUTABLE_READY)));
 
+  g_object_class_install_property (gobject_class, PROP_LOG_SEVERITY,
+      g_param_spec_enum ("log-severity", "log-severity",
+          "CEF log severity level ",
+          GST_TYPE_CEF_LOG_SEVERITY_MODE, DEFAULT_LOG_SEVERITY,
+          (GParamFlags) (G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS)));
```
[the remark this answers is not in the document]
Comment (aiden-loroco) [reply gstcefsrc.cc]: Thanks!

**Comment by MathieuDuponchelle** [comment]
All good now let's go, thanks :muscle:

**Pull request merged**